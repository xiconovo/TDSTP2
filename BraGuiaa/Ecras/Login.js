import React, { useState } from 'react';
import { StyleSheet, ImageBackground, TextInput, Button } from 'react-native';
import { loginUser } from '../Api/api';

const Login = ({navigation}) => {
    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');

    const handleLogin = () => {
        console.log('Button pressed');
        loginUser(username, password, navigation);
    };

    return (
        <ImageBackground source={require('../assets/braga2.jpeg')} style={styles.container}>
            <TextInput
                style={styles.input}
                onChangeText={setUsername}
                value={username}
                placeholder="Username"
            />
            <TextInput
                style={styles.input}
                onChangeText={setPassword}
                value={password}
                placeholder="Password"
                secureTextEntry
            />
            <Button
                title="Login"
                onPress={handleLogin}
            />
        </ImageBackground>
    );
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'center',
    },
    input: {
        height: 40,
        width: '80%',
        borderColor: 'gray',
        borderWidth: 1,
        marginBottom: 10,
        padding: 10,
        backgroundColor: 'rgba(255,255,255,0.8)', // Para dar um fundo semi-transparente ao input
    },
});

export default Login;
