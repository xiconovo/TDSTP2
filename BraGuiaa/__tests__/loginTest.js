import Login from '../Ecras/Login';
import React from 'react';
import renderer from 'react-test-renderer';




test('renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
});