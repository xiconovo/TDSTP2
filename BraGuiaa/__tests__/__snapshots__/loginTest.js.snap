// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/braga2.jpeg",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Username"
    style={
      {
        "backgroundColor": "rgba(255,255,255,0.8)",
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
        "width": "80%",
      }
    }
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      {
        "backgroundColor": "rgba(255,255,255,0.8)",
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 10,
        "padding": 10,
        "width": "80%",
      }
    }
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
